src/pudl_archiver/frictionless.py

```diff
@@ -177,30 +183,29 @@ def from_pudl_metadata(
         )
 
     @classmethod
-    def mecs(cls, resources: Iterable[Resource], version: str | None):
-        """Hack method to create a Datapackage for EIA MECS data not in PUDL metadata."""
+    def from_non_pudl_metadata(
+        cls,
+        name: str,
+        resources: Iterable[Resource],
+        version: str | None,
+    ):
+        """Create a datapackage for sources that won't end up in PUDL."""
+        # TODO: This is a slightly ugly workaround to avoid having to add sources into
+        # the from_id method - should I just fix this at the source?
+        data_source = DataSource(
```

Comment:
I think if you change this to call `from_id` instead of `dict_from_id`, that *should* fix the broken tests.

Reply:
Whoops, I see now that `from_id` doesn't take the `sources` dict right now. It's a bit of a pain to have to do another PUDL PR, but I think the most consistent API would be to add `sources` to the `from_id` method, since you're basically just copying what `from_id` does here.